props/saber_BC_buttons.h

```diff
@@ -477,738 +1491,1257 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+  bool isSecondBladeOn() { return SaberBase::OnBlades()[BC_SECOND_BLADE]; }
+
+  // Get blade above horizon line to apply effect to
+  BladeSet GetEffectTargetBlade(bool return_opposite = false) {
+      if (!return_opposite) {
+          return accel_.x > 0 ? BC_MAIN_BLADE_SET : BC_SECOND_BLADE_SET;
       } else {
-        VolumeDown();
+          return accel_.x > 0 ? BC_SECOND_BLADE_SET : BC_MAIN_BLADE_SET;
       }
-      return true;
+  }
 
-// Start or Stop Track
-#if NUM_BUTTONS == 1
-    case EVENTID(BUTTON_POWER, EVENT_FOURTH_SAVED_CLICK_SHORT, MODE_OFF):
-    case EVENTID(BUTTON_POWER, EVENT_FOURTH_SAVED_CLICK_SHORT, MODE_ON):
-#else
-    // 2 or 3 button
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_ON | BUTTON_AUX):
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_OFF | BUTTON_AUX):
+  void TurnBladeOn(BladeSet target_blade) {
+    // Add in all non-controlled blades, effectively excluding the "other" blade.
+    target_blade = target_blade | ~controlled_blades_;
+    if (SaberBase::OnBlades().off()) {
+      PVLOG_NORMAL << "** No blades are currently ON, turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade and all others, excluding the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "SECOND" : "MAIN")
+                   << " blade\n";
+      if (isPointingUp() || muted_) {
+        FastOn(EffectLocation(0, target_blade));
+     } else {
+        On(EffectLocation(0, target_blade));
+      }
+    } else {
+      PVLOG_NORMAL << "** Turning on the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND")
+                   << " blade\n";
+      SaberBase::TurnOn(EffectLocation(0, target_blade));
+    }
+  }
+
+  void TurnBladeOff(BladeSet target_blade) {
+    SaberBase::OffType off_type = isPointingUp() ? SaberBase::OFF_FAST : SaberBase::OFF_NORMAL;
+    PVLOG_DEBUG << "***** Off type: " << (off_type == SaberBase::OFF_FAST ? "OFF_FAST" : "OFF_NORMAL") << "\n";
+
+    // Check if this is the only blade ON (of MAIN or SECOND blades)
+    if ((SaberBase::OnBlades() & ~target_blade & controlled_blades_).off()) {
+      // Other blade is not on, so just do normal Off() with appropriate off_type
+      PVLOG_NORMAL << "** Turning OFF all blades\n";
+      Off(off_type);
+      muted_ = false;
+    } else {
+      // Only Turn OFF this blade, leave the other one ON.
+      PVLOG_NORMAL << "** Turning OFF only the " 
+                   << (target_blade[BC_MAIN_BLADE] ? "MAIN" : "SECOND") 
+                   << " blade\n";
+      SaberBase::TurnOff(off_type, EffectLocation(0, target_blade));
+    }
+    battle_mode_ = false;
+    spam_blast_ = false;
+    saber_off_time_ = millis();
+  }
+
+  void TurnMainBladeOnMuted() {
+    if (hybrid_font.PlayPolyphonic(&SFX_mute)) {
+      mute_mainBlade_timer_.trigger(SaberBase::sound_length * 1000);
+    } else {
+      if (SetMute(true)) {
```

Comment:
This code is duplicated here and in loop().
Duplicated code is generally a Bad Thing (tm).
It seems easy to avoid here though, as you could just always call mute_mainBlade_timer_.trigger().
If SFX_mute doesn't exist, then just call it with a delay of zero, right?

Reply:
oh that's great, yes.
Changed for all 3 mute timers (all, main, and second)